### ISSUE TITLE
NameError: name '_C' is not defined

### ISSUE THREAD

**Issue by g558800** [IDEA-Research/Grounded-Segment-Anything#169]
May I ask if this repo requires the installation of VC++buildtools?

**Comment by g558800** [comment]
May I use the torch-2.0.0+cu118 as the CUDA_HOME?It is include cuda toolkit 118?

**Comment by souvik0306** [comment]
Having the same issue. Did anyone resolve it?

**Comment by TestPrab** [comment]
> May I use the torch-2.0.0+cu118 as the CUDA_HOME?It is include cuda toolkit 118?

When you build custom cuda extension that is `_C` file , it uses local cuda toolkit, hence you need to install cuda_toolkit 11.8 in your system , and then set CUDA_HOME to the location where it's installed , along with that also set up the the LD_LIBRARY  https://forums.developer.nvidia.com/t/path-ld-library-path/48080

**Comment by g558800** [comment]
Can I install the latest cudatoolkit 12.1 directly, even if I'm using cu118?

**Comment by TestPrab** [comment]
> 

No, when building custom cuda extension local cuda toolkit should match to the pytorch's cuda support version , hence you need to install cuda 11.8 version  
https://developer.nvidia.com/cuda-11-8-0-download-archive

**Comment by g558800** [comment]
By the way,I have other repo using cu117 etc, do I need to install more than one cudatoolkit 11.7 in addition?

**Comment by TestPrab** [comment]
Not sure if you can several cuda versions or not, but yess the cuda version in local must match your pytorch version only for the cases where you build **custom cuda extension** , else no need.
